tools/ja-rule/ja-rule.cpp

```diff
@@ -50,527 +57,511 @@ using ola::rdm::RDMDiscoveryResponse;
-  }
-
   void ResetDevice() {
-    if (!CheckForDevice()) {
+    if (!CheckForWidget()) {
       return;
     }
 
-    m_device->SendMessage(JaRuleEndpoint::RESET_DEVICE, NULL, 0);
+    m_widget->SendCommand(
+        FLAGS_port,
+        ola::usb::JARULE_CMD_RESET_DEVICE,
+        NULL, 0,
+        NewSingleCallback(this, &Controller::AckCommandComplete));
   }
 
   void GetFlags() {
-    if (!CheckForDevice()) {
+    if (!CheckForWidget()) {
       return;
     }
 
-    m_device->SendMessage(JaRuleEndpoint::GET_FLAGS, NULL, 0);
+    m_widget->SendCommand(
+        FLAGS_port,
+        ola::usb::JARULE_CMD_GET_FLAGS,
+        NULL, 0,
+        NewSingleCallback(this, &Controller::GetFlagsCommandComplete));
   }
 
   void _SendDMX(const uint8_t *data, unsigned int size) {
-    if (!CheckForDevice()) {
+    if (!CheckForWidget()) {
       return;
     }
-    m_device->SendMessage(JaRuleEndpoint::TX_DMX, data, size);
+    m_widget->SendCommand(
+        FLAGS_port,
+        ola::usb::JARULE_CMD_TX_DMX,
+        data, size,
+        NewSingleCallback(this, &Controller::CommandComplete));
   }
 
   void Adjust(bool increase) {
-    if (m_mode == EDIT_BREAK) {
-      if (increase) {
-        m_break++;
-      } else {
-        m_break--;
+    TimingSetting& setting = m_timing_settings[m_current_timing_option];
+    if (increase) {
+      setting.current_value++;
+      if (setting.current_value > setting.max_value) {
+        setting.current_value = setting.max_value;
       }
-      cout << "Break is now " << m_break << " us" << endl;
-    } else if (m_mode == EDIT_MAB) {
-      if (increase) {
-        m_mab++;
-      } else {
-        m_mab--;
-      }
-      cout << "MAB is now " << m_mab << " us" << endl;
-    } else if (m_mode == EDIT_RDM_WAIT_TIME) {
-      if (increase) {
-        m_rdm_wait_time++;
-      } else {
-        m_rdm_wait_time--;
-      }
-      float wait_time = m_rdm_wait_time / 10.0;
-      cout << "RDM Wait time is now " << wait_time << " ms" << endl;
-    } else if (m_mode == EDIT_RDM_BROADCAST_LISTEN) {
-      if (increase) {
-        m_rdm_broadcast_listen++;
-      } else {
-        m_rdm_broadcast_listen--;
+    } else {
+      setting.current_value--;
+      if (setting.current_value < setting.min_value) {
+        setting.current_value = setting.min_value;
       }
-      float listen_time = m_rdm_broadcast_listen / 10.0;
-      cout << "RDM Broadcast listen is now " << listen_time << " ms"
-           << endl;
     }
+
+    cout << setting.description << " is now ";
+    if (setting.units == TIMING_UNITS_MICROSECONDS) {
+      cout << setting.current_value << " us";
+    } else if (setting.units == TIMING_UNITS_TENTHS_OF_MILLI_SECONDS) {
+      float adjusted_time = setting.current_value / 10.0;
+      cout << adjusted_time << " ms";
+    }
+    cout << endl;
   }
 
   void Commit() {
-    if (!CheckForDevice()) {
+    if (!CheckForWidget()) {
       return;
     }
 
+    if (m_mode == MODE_DEFAULT) {
```

Comment:
I'd disagree, commit is only known to be valid in edit mode currently. If I add another mode that just keeps fetching some counters or something, commit won't be relevant there. Perhaps we should change the error message, but it should default to only allowing commit in defined modes, rather than the other way round.